src/provider/provider_level_zero.c

```diff
@@ -205,7 +226,9 @@ static umf_result_t ze_memory_provider_alloc(void *provider, size_t size,
     case UMF_MEMORY_TYPE_DEVICE: {
         ze_device_mem_alloc_desc_t dev_desc = {
             .stype = ZE_STRUCTURE_TYPE_HOST_MEM_ALLOC_DESC,
```

Comment:
This should probably also be a `ZE_STRUCTURE_TYPE_DEVICE_MEM_ALLOC_DESC`.

Reply:
Done.